### PR TITLE
reservations: check if channel is closing

### PR DESCRIPTION
**Pull request by sputn1ck**
This PR adds a check to the runtime loop
of the reservation manager to check if the
channel is closing before trying to read
from it.

### CODE REVIEW

**Review by bhandras — APPROVED** [review]

File: instantout/reservation/manager.go [hunk 1]
```diff
@@ -64,7 +64,12 @@ func (m *Manager) Run(ctx context.Context, height int32) error {
 			log.Debugf("Received block %v", height)
 			currentHeight = height
 
-		case reservationRes := <-ntfnChan:
+		case reservationRes, ok := <-ntfnChan:
+			if !ok {
+				// The channel has been closed, we'll stop the
+				// reservation manager.
```
Comment (starius) [inline instantout/reservation/manager.go:70]: Should we add the logging here like in `runCtx.Done` case?

```go
log.Debugf("Stopping reservation manager (ntfnChan closed)")
```

File: instantout/reservation/manager.go [hunk 2]
```diff
@@ -56,15 +56,21 @@ func (m *Manager) Run(ctx context.Context, height int32) error {
...
+				log.Debugf("Stopping reservation manager (ntfnChan closed)")
+				return nil
+			}
```
Comment (bhandras) [inline instantout/reservation/manager.go:73]: nit: add extra nl below

File: instantout/reservation/manager.go [hunk 3]
```diff
@@ -56,15 +56,22 @@ func (m *Manager) Run(ctx context.Context, height int32) error {
...
 		case height := <-newBlockChan:
 			log.Debugf("Received block %v", height)
 			currentHeight = height
 
-		case reservationRes := <-ntfnChan:
+		case reservationRes, ok := <-ntfnChan:
+			if !ok {
+				// The channel has been closed, we'll stop the
+				// reservation manager.
+				log.Debugf("Stopping reservation manager (ntfnChan closed)")
```
Comment (starius) [inline instantout/reservation/manager.go:71]: add newline before `return`

**Review by starius — APPROVED** [review]
LGTM 🏆

**Pull request merged**